packages/angular_devkit/build_angular/src/utils/purge-cache.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { BuilderContext } from '@angular-devkit/architect';
+import { PathLike, existsSync, promises as fsPromises } from 'fs';
+import { join } from 'path';
+import { normalizeCacheOptions } from './normalize-cache';
+
+/** Delete stale cache directories used by previous versions of build-angular. */
+export async function purgeStaleBuildCache(context: BuilderContext): Promise<void> {
+  const projectName = context.target?.project;
+  if (!projectName) {
+    return;
+  }
+
+  const metadata = await context.getProjectMetadata(projectName);
+  const { basePath, path, enabled } = normalizeCacheOptions(metadata, context.workspaceRoot);
+
+  if (!enabled || !existsSync(basePath)) {
+    return;
+  }
+
+  const entries = await fsPromises.readdir(basePath);
+
+  for (const entry of entries) {
```

Comment:
Would it be work parallelizing all the deletes instead of running them sequentially?

Reply:
That should work too.